siliconcompiler/tools/openroad/sc_floorplan.tcl

```diff
@@ -107,6 +113,27 @@ if {[expr ! [dict exists $sc_cfg "input" "floorplan.def"]]} {
     ###########################
     set def [dict get $sc_cfg "input" "floorplan.def"]
     read_def -floorplan_initialize $def
+
+    if [dict exists $sc_cfg pdk $sc_pdk aprtech openroad $sc_stackup $sc_libtype macroplace] {
+        ###########################
+        # TDMS Placement
+        ###########################
+        global_placement -density $openroad_place_density \
+            -pad_left $openroad_pad_global_place \
+            -pad_right $openroad_pad_global_place
+
+        ###########################
+        # Manual macro placement
+        ###########################
+        source [lindex [dict get $sc_cfg pdk $sc_pdk aprtech openroad $sc_stackup $sc_libtype macroplace] 0]
+    }
```

Comment:
I just realized that I mixed up the "floorplan DEF does/doesn't exist" branches, so it would probably be okay to delete this block; let me know what you think.